src/main/java/com/exasol/parquetio/data/GenericRow.java

```diff
@@ -0,0 +1,91 @@
+package com.exasol.parquetio.data;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * An implementation of {@link Row} that uses unmodifiable list for storing values.
+ */
+public class GenericRow implements Row {
+    private static final long serialVersionUID = -6175254671958738095L;
+
+    private final List<Object> values;
+
+    /**
+     * A constructor to create a new instance.
+     *
+     * @param values list of values
+     */
+    public GenericRow(final List<Object> values) {
+        this.values = Collections.unmodifiableList(values);
+    }
+
+    /**
+     * A factory method to create a new instance.
+     *
+     * @param values list of values
+     */
+    public static GenericRow of(final Object... values) {
+        return new GenericRow(Arrays.asList(values));
+    }
+
+    @Override
+    public Object getObjectAt(final int position) {
+        return values.get(position);
+    }
+
+    @Override
+    public List<Object> getValues() {
+        return values;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (!(other instanceof GenericRow)) {
+            return false;
+        }
+        GenericRow otherRow = (GenericRow) other;
+        if (this.size() != otherRow.size()) {
+            return false;
+        }
+        for (int i = 0; i < this.size(); i++) {
```

Comment:
From the List.equals javadoc:

    Returns true if and only if the specified object is also a list, both lists have the same size, and all corresponding pairs of elements in the two lists are equal. (Two elements e1 and e2 are equal if (e1==null ? e2==null : e1.equals(e2)).)

So maybe you can simply do `this.values.equals(other.values)`

Reply:
I would expect this to be a lot faster.

A good reason for your solution would be if you want to check equality against the `Row` interface.
But in that case, the `if` in line 49 should check for `Row` instead of `GenericRow`.

Reply:
and you could maybe move the whole function as a default to the `Row` interface...

But since I would expect that it has a bad performance maybe its better to leave it here and use the equals on the list

Reply:
Yes, that nice find! Line 49 should be Row definitely.